### PR TITLE
Bigquery dataset routines

### PR DESCRIPTION
**Pull request by apichick**
Added routines to bigquery-dataset module

---
**Checklist**

I applicable, I acknowledge that I have:
- [X] Read the [contributing guide](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md)
- [X] Ran `terraform fmt` on all modified files
- [X] Regenerated the relevant README.md files using [`tools/tfdoc.py`](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md#fabric-tools)
- [X] Made sure all relevant tests pass

### CODE REVIEW

File: modules/bigquery-dataset/variables.tf [hunk 1]
```diff
@@ -164,6 +164,44 @@ variable "project_id" {
...
+    imported_libraries   = optional(list(string))
+    determinism_level    = optional(string)
+    data_governance_type = optional(string)
+    return_table_type    = optional(string)
+    arguments = optional(map(object({
+      argument_kind = optional(string)
+      mode          = optional(string)
+      data_type     = optional(string)
+    })), {})
+    spark_options = optional(list(object({
```
Comment (wiktorn) [inline modules/bigquery-dataset/variables.tf:183]:
```suggestion
    spark_options = optional(object({
```
Does providing multiple `spark_options` really make sense?

File: modules/bigquery-dataset/variables.tf [hunk 2]
```diff
@@ -164,6 +164,44 @@ variable "project_id" {
...
+      connection      = optional(string)
```
Comment (wiktorn) [inline modules/bigquery-dataset/variables.tf:184]: From quickly skimming the [docs](https://cloud.google.com/bigquery/docs/spark-procedures) I presume that `connection` is not optional

**Review by wiktorn — APPROVED** [review]
Looks good, few nits here and there.

File: modules/bigquery-dataset/variables.tf [hunk 3]
```diff
@@ -164,6 +164,44 @@ variable "project_id" {
...
+      container_image = optional(string)
+      properties      = optional(map(string))
+      main_file_uri   = optional(string)
+      py_file_uris    = optional(list(string))
+      jar_uris        = optional(list(string))
+      file_uris       = optional(list(string))
+      archive_uris    = optional(list(string))
+      main_class      = optional(string)
+    })), [])
+    remote_function_options = optional(list(object({
```
[the remark this answers is not in the document]
Comment (apichick) [reply modules/bigquery-dataset/variables.tf]: true, my mistake, spark options was also wrong, just an object

File: modules/bigquery-dataset/variables.tf [hunk 4]
```diff
@@ -164,6 +164,44 @@ variable "project_id" {
...
+      endpoint             = optional(string)
+      connection           = optional(string)
+      user_defined_context = optional(map(string))
+      max_batching_rows    = string
```
[the remark this answers is not in the document]
Comment (apichick) [reply modules/bigquery-dataset/variables.tf]: done

File: modules/bigquery-dataset/variables.tf [hunk 5]
```diff
@@ -164,6 +164,44 @@ variable "project_id" {
...
+      main_file_uri   = optional(string)
+      py_file_uris    = optional(list(string))
+      jar_uris        = optional(list(string))
+      file_uris       = optional(list(string))
+      archive_uris    = optional(list(string))
+      main_class      = optional(string)
+    })), [])
+    remote_function_options = optional(list(object({
+      endpoint             = optional(string)
+      connection           = optional(string)
```
[the remark this answers is not in the document]
Comment (apichick) [reply modules/bigquery-dataset/variables.tf]: done, spark options was also not ok

**Pull request merged**